lib/index.ts

```diff
@@ -0,0 +1,68 @@
+import * as fs from 'fs';
+import * as path from 'path';
+
+import { InvalidUserInputError } from './errors';
+import { FileParser } from './parsers/file-parser';
+import { ComposerParser } from './parsers/composer-parser';
+import { ComposerJsonFile, ComposerLockFile, ComposerParserResponse, SystemPackages } from './types';
+
+function buildDepTree(
+    lockFileContent: string,
+    manifestFileContent: string,
+    defaultProjectName: string,
+    systemVersions: SystemPackages): ComposerParserResponse {
+
+    const lockFileJson: ComposerLockFile = FileParser.parseLockFile(lockFileContent);
+    const manifestJson: ComposerJsonFile = FileParser.parseManifestFile(manifestFileContent);
+
+    if (typeof lockFileJson.packages !== 'object') {
+        throw new InvalidUserInputError('Invalid lock file. Must contain `packages` property');
+    }
+
+    const applicationName: string = manifestJson.name || defaultProjectName;
+    const applicationVersion: string = ComposerParser.getVersion(manifestJson) || '0.0.0';
+
+    return {
+        name: applicationName,
+        version: applicationVersion,
+        packageFormatVersion: 'composer:0.0.1',
+        dependencies: ComposerParser.buildDependencies(manifestJson, lockFileJson.packages, manifestJson,
+            [`${applicationName}@${applicationVersion}`], systemVersions),
+    };
+}
+
+function buildDepTreeFromFiles(
+    basePath: string,
+    lockFileName: string,
+    systemVersions: SystemPackages): ComposerParserResponse {
+    if (!basePath || !lockFileName || !systemVersions) {
+        throw new InvalidUserInputError('Missing required parameters for buildDepTreeFromFiles()');
```

Comment:
I'd rather print the name of the missing parameter (so, 3 different checks I guess?)